nix/marlowe-symbolic-lambda.nix

```diff
@@ -0,0 +1,41 @@
+{ pkgs, haskellPackages }:
+let
+  # NOTE: Unfortunately we cannot build a static executable if the package uses template haskell.
+  # In order to get around this we make sure that the TH code is all in libraries and create
+  # a single haskell file that does nothing other than import the main function from the library.
+  # We can then statically compile this as it does not use TH.
+  staticHaskellPackages = (haskellPackages.override (old: {
```

Comment:
I'd be tempted to just use the argument `haskellPackages` and then pass this in as the argument. Otherwise this now reads a bit weird.

I'm going by a heuristic here that "cross-compiled" packages (which static packages sort of are) should "look like" normal packages but just be called differently.